ci-automation/vendor-testing/openstack.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+# Copyright (c) 2022 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+# Test execution script for the Openstack vendor image.
+# This script is supposed to run in the mantle container.
+
+source ci-automation/vendor_test.sh
+
+# OPENSTACK_CREDS, OPENSTACK_USER, OPENSTACK_HOST, OPENSTACK_KEYFILE should be provided by sdk_container/.env
+config_file=''
+secret_to_file config_file "${OPENSTACK_CREDS}"
+
+openstack_keyfile=''
+secret_to_file openstack_keyfile "${OPENSTACK_KEYFILE}"
+
+# Upload the image on OpenStack dev instance.
```

Comment:
Does this overwrite the image if it exists as leftover or do we need a deletion action before running this, just in case?

We also need to delete the image at the end (via a trap), and extend `ci-automation/garbage_collect_cloud.sh` to delete old images and old instances.

Reply:
I checked - uniqueness is ensure via the image ID and not the name.

```bash
$ openstack image list
+--------------------------------------+--------------------------+--------+
| ID                                   | Name                     | Status |
+--------------------------------------+--------------------------+--------+
| e1f00e71-ccd3-4e89-b28b-44ce310bad49 | cirros-0.5.2-x86_64-disk | active |
| 11fa2619-1437-42ed-91c2-dd9a0d630bfc | flatcar-lts-3033         | active |
| 912f2af6-e448-4b1b-b130-2d52718d1e10 | flatcar-lts-3033         | active |
| e4244cba-b2c9-44e1-92bb-acea2aaaa35d | flatcar-lts-3033         | active |
+--------------------------------------+--------------------------+--------+
```

So we're good on this side. We just need to add the image deletion via a trap command and extend the GC as you suggested.